internal/network/router/apphandler.go

```diff
@@ -0,0 +1,115 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"parabellum.kproducer/internal/config"
+	"parabellum.kproducer/internal/model"
+	"parabellum.kproducer/internal/network/communicator"
+	"parabellum.kproducer/internal/network/server"
+	"parabellum.kproducer/internal/pubsub"
+
+	"github.com/go-chi/chi/v5"
+)
+
+type AppHandler struct {
+	Producer   *pubsub.Producer
+	GrpcClient *communicator.ClientGRPC
+	HttpServer *server.HTTP
+	router     chi.Router
+	ctx        context.Context
```

Comment:
Pls, take a look [Contexts and structs](https://go.dev/blog/context-and-structs)